### ISSUE TITLE
not correct logic route decorator when miltiple time register same router

### ISSUE THREAD

**Issue by mmmcorpsvit** [tiangolo/fastapi#4422]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
#
# @router.get("/api/test")
#  def ....

# #then write bootom
# @router.get("/api/test", response_model=test_model)
#  def ....
#     return {} # <- there just static correct placeholder which equal response model
```

### Description

in this case execute first def, but in swagger will be second params list and response model

how prevent same router?
maybe need detect this cases?

### Operating System

Windows

### FastAPI Version

0.63.0

### Python Version

3.9

**Comment by mmmcorpsvit** [comment]
detect:
```python
from fastapi import Request
@router.get('/list_endpoints/')
def list_endpoints(request: Request):
    url_list = []
    bad_url_list = []
    for i in request.app.routes:
        for j in i.methods:
            if {'path': i.path, 'name': i.name, 'methods': j} in url_list:
                bad_url_list.append({'path': i.path, 'name': i.name, 'methods': j})

            url_list.append({'path': i.path, 'name': i.name, 'methods': j})

    return bad_url_list

```

**Comment by rafsaf** [comment]
You are right, but I doubt you can miss that and I don't think you need any additional code to check it.


That often happens when you duplicate endpoint, but then IDE screams and curses in pain seeing doubled `def xyz` function name, your unit or integration tests won't pass with any luck and there should be also code review. Also unless you specify "name" in `router.get(...)`, it is equal to function name. Good practise is to use **different** routers in different files with different prefix - "auth", "user", "pony", "mickey-mouse" - you name it, should be perfectly enough.

**Comment by mmmcorpsvit** [comment]
this check can be implemented in CI/CD
in real world with enterprise/legacy/no my code this check just need

**Comment by Pacheco95** [comment]
Why does FastAPI don't replace the old router by the new one? I think this beahaviour is at least reasonable.
